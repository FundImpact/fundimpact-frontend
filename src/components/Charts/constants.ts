export const options = {
	legend: {
		display: false,
	},
	tooltips: {
		enabled: false,
	},
	hover: false,
};
