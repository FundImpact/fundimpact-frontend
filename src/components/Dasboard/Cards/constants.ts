export enum CARD_TYPES {
	PROJECT = "PROJECT",
	PROGRESS = "PROGRESS",
	PIE = "PIE",
	DEFAULT = "DEFAULT",
}

export enum CARD_OF {
	DELIVERABLE = "DELIVERABLE",
	BUDGET = "BUDGET",
	IMPACT = "IMPACT",
}
