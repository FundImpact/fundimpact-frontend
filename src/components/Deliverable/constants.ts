export enum DELIVERABLE_ACTIONS {
	CREATE = "CREATE",
	UPDATE = "UPDATE",
}
