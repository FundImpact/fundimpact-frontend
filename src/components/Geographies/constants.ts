export enum GEOGRAPHIES_ACTIONS {
	CREATE = "CREATE",
	UPDATE = "UPDATE",
}
