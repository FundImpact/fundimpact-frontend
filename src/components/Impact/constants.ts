export enum IMPACT_ACTIONS {
	CREATE = "CREATE",
	UPDATE = "UPDATE",
}
