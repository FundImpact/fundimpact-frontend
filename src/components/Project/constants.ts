export enum PROJECT_ACTIONS {
	CREATE = "CREATE",
	UPDATE = "UPDATE",
}
