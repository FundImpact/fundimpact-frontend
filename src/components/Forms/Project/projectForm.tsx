import {
	Button,
	createStyles,
	Grid,
	TextField,
	Theme,
	FormControl,
	InputLabel,
	Select,
	Box,
	MenuItem,
	Dialog,
	DialogContent,
	Typography,
	Card,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { Form, Formik } from "formik";
import { IProjectFormProps } from "../../../models/project/ProjectForm";
import { PROJECT_ACTIONS } from "../../Project/constants";
import { IProject, ProjectProps } from "../../../models/project/project";
import React, { useState } from "react";
import Slide from "@material-ui/core/Slide";
import { TransitionProps } from "@material-ui/core/transitions";

const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		root: {
			display: "flex",
			flexDirection: "column",
			"& .MuiTextField-root,": {
				margin: theme.spacing(1),
			},
		},
		button: {
			color: theme.palette.common.white,
			margin: theme.spacing(1),
			marginLeft: theme.spacing(0),
		},
		leftBox: {
			width: "100%",
			backgroundColor: "#e3f2fd",
			height: "40%",
			marginTop: theme.spacing(1),
		},
		form: {
			margin: theme.spacing(1),
		},
		formControl: {
			margin: theme.spacing(1),
			width: "100%",
		},
	})
);

const Transition = React.forwardRef(function Transition(
	props: TransitionProps & { children?: React.ReactElement<any, any> },
	ref: React.Ref<unknown>
) {
	return <Slide direction="up" ref={ref} {...props} />;
});

function ProjectForm({
	clearErrors,
	initialValues,
	validate,
	formState,
	onCreate,
	onUpdate,
	children,
	workspaces,
	formIsOpen,
	handleFormOpen,
}: IProjectFormProps & React.PropsWithChildren<IProjectFormProps>) {
	const classes = useStyles();

	const validateInitialValue = (initialValue: IProject) => {
		const errors = validate(initialValue) as object;
		if (!errors) return true;
		return Object.keys(errors).length ? false : true;
	};
	return (
		<Dialog
			fullWidth
			open={formIsOpen}
			aria-labelledby="form-dialog-title"
			maxWidth="md"
			TransitionComponent={Transition}
		>
			<DialogContent>
				<Box
					mx="auto"
					height={"100%"}
					width={{ xs: "100%", md: "100%", lg: "100%" }}
					onChange={clearErrors}
				>
					<Grid container justify={"center"}>
						<Grid item xs={4}>
							<Typography variant="h6" gutterBottom>
								New Project
							</Typography>
							<Typography variant="subtitle2" gutterBottom color="textSecondary">
								Add a project to start tracking your funds and impact
							</Typography>
							<Card elevation={0} className={classes.leftBox}>
								<Box mt={2} ml={3}>
									<Typography variant="subtitle1" gutterBottom color="primary">
										Organisation 1
									</Typography>
								</Box>
								<Box m={1} ml={3}>
									<Typography variant="body2" gutterBottom color="textPrimary">
										Workspace 1
									</Typography>
								</Box>
							</Card>
						</Grid>
						<Grid item xs={8}>
							<div onChange={clearErrors} className={classes.form}>
								<Formik
									initialValues={initialValues}
									validate={validate}
									isInitialValid={(props: any) =>
										validateInitialValue(props.initialValues)
									}
									onSubmit={(values) =>
										formState === PROJECT_ACTIONS.CREATE
											? onCreate(values)
											: onUpdate(values)
									}
								>
									{(formik) => {
										return (
											<Form
												id="project_form"
												className={classes.root}
												autoComplete="off"
											>
												<Grid container spacing={1} justify={"center"}>
													<TextField
														value={formik.values.name}
														error={!!formik.errors.name}
														helperText={
															formik.touched.name &&
															formik.errors.name
														}
														onChange={formik.handleChange}
														data-testid="createProjectName"
														label="Project Name"
														inputProps={{
															"data-testid": "createProjectNameInput",
														}}
														required
														fullWidth
														name="name"
														variant="outlined"
													/>
													<TextField
														onChange={formik.handleChange}
														data-testid="createProjectShortName"
														inputProps={{
															"data-testid":
																"createProjectShortNameInput",
														}}
														value={formik.values.short_name}
														label="Short Name"
														fullWidth
														name="short_name"
														variant="outlined"
													/>
													<FormControl
														variant="outlined"
														className={classes.formControl}
													>
														<InputLabel id="demo-simple-select-outlined-label">
															Choose Workspace
														</InputLabel>
														{workspaces && (
															<Select
																labelId="demo-simple-select-outlined-label"
																id="demo-simple-select-outlined"
																error={!!formik.errors.workspace}
																value={formik.values.workspace}
																onChange={formik.handleChange}
																label="Choose workspace"
																name="workspace"
																data-testid="createProjectWorkspace"
																inputProps={{
																	"data-testid":
																		"createProjectWorkspaceOption",
																}}
															>
																{workspaces &&
																	workspaces.map(
																		(
																			elem: {
																				id: number;
																				name: string;
																			},
																			index: number
																		) => (
																			<MenuItem
																				key={index}
																				value={elem.id}
																			>
																				{elem.name}
																			</MenuItem>
																		)
																	)}
															</Select>
														)}
													</FormControl>
													<TextField
														data-testid="createProjectDescription"
														value={formik.values.description}
														onChange={formik.handleChange}
														inputProps={{
															"data-testid":
																"createProjectDescriptionInput",
														}}
														label="Description.. ( Optional )"
														multiline
														rows={3}
														name="description"
														type="text"
														variant="outlined"
														fullWidth
													/>
												</Grid>
												<Box display="flex" m={1}>
													<Button
														color="secondary"
														className={classes.button}
														onClick={handleFormOpen}
														variant="contained"
													>
														Cancel
													</Button>
													<Button
														className={classes.button}
														data-testid="createProjectSubmit"
														disabled={!formik.isValid}
														form="project_form"
														type="submit"
														color="primary"
														variant="contained"
													>
														{formState === PROJECT_ACTIONS.CREATE
															? "Create"
															: "Update"}
													</Button>
												</Box>
											</Form>
										);
									}}
								</Formik>
							</div>
						</Grid>
					</Grid>
				</Box>
			</DialogContent>
			{children ? children : null}
		</Dialog>
	);
}

export default ProjectForm;
