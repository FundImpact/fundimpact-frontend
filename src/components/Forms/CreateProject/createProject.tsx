import {
	Button,
	createStyles,
	Grid,
	TextField,
	Theme,
	FormControl,
	InputLabel,
	Select,
	Box,
	MenuItem,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { Form, Formik } from "formik";
import { IProjectFormProps } from "../../../models/project/ProjectForm";
import { PROJECT_ACTIONS } from "../../Project/constants";
import React from "react";

// import { useNavigate } from 'react-router-dom';

const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		form: {
			display: "flex",
			"& .MuiTextField-root": {
				margin: theme.spacing(1),
				spacing: 1,
			},
			"& .MuiOutlinedInput-input": {
				padding: theme.spacing(2),
			},
		},
		formControl: {
			margin: theme.spacing(1),
			width: "100%",
		},
	})
);

function CreateProject({
	clearErrors,
	initialValues,
	validate,
	formState,
	onCreate,
	onUpdate,
	children,
	workspace,
}: IProjectFormProps & React.PropsWithChildren<IProjectFormProps>) {
	const classes = useStyles();
	return (
		<div onChange={clearErrors}>
			<Formik
				initialValues={initialValues}
				onSubmit={(values) =>
					formState === PROJECT_ACTIONS.CREATE ? onCreate(values) : onUpdate(values)
				}
			>
				{(formik) => {
					return (
						<Form className={classes.form}>
							<Grid container spacing={1} justify={"center"}>
								<Grid item xs={6} md={6}>
									<TextField
										error={!!formik.errors.name}
										helperText={formik.touched.name && formik.errors.name}
										onChange={formik.handleChange}
										label="Project Name"
										required
										fullWidth
										name="name"
										variant="outlined"
									/>
								</Grid>
								<Grid item xs={6} md={6}>
									<TextField
										error={!!formik.errors.short_name}
										helperText={
											formik.touched.short_name && formik.errors.short_name
										}
										onChange={formik.handleChange}
										label="Short Name"
										required
										fullWidth
										name="short_name"
										variant="outlined"
									/>
								</Grid>
								<Grid item xs={12} md={12}>
									<FormControl variant="outlined" className={classes.formControl}>
										<InputLabel id="demo-simple-select-outlined-label">
											Choose Workspace
										</InputLabel>
										{workspace && (
											<Select
												labelId="demo-simple-select-outlined-label"
												id="demo-simple-select-outlined"
												value={formik.values.workspace}
												onChange={formik.handleChange}
												label="Choose workspace"
												name="workspace"
											>
												{workspace &&
													workspace.map((elem: any, index: number) => (
														<MenuItem key={index} value={elem.id}>
															{elem.name}
														</MenuItem>
													))}
											</Select>
										)}
									</FormControl>
								</Grid>
								<Grid xs={12}>
									<TextField
										data-testid="description"
										value={formik.values.description}
										error={!!formik.errors.description}
										onChange={formik.handleChange}
										label="Description.. ( Optional )"
										multiline
										rows={3}
										name="description"
										type="text"
										variant="outlined"
										fullWidth
									/>
								</Grid>
								<Grid item container>
									<Grid xs={8}></Grid>
									<Grid xs={4}>
										<Box mt={3}>
											<Button
												fullWidth
												disabled={!formik.isValid}
												type="submit"
												variant="contained"
												color="primary"
											>
												Submit
											</Button>
										</Box>
									</Grid>
								</Grid>
							</Grid>
						</Form>
					);
				}}
			</Formik>
		</div>
	);
}

export default CreateProject;
