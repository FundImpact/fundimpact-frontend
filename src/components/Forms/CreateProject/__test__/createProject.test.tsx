import React from "react";
import ReactDOM from "react-dom";
import CreateProject from "./../createProject";
import {
	act,
	fireEvent,
	queries,
	render,
	RenderResult,
	wait,
	cleanup,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { IProject } from "../../../../models/project/project";
import { PROJECT_ACTIONS } from "../../../Project/constants";

const intialFormValue: IProject = {
	id: 1,
	name: "Project1",
	short_name: "P1",
	description: "",
	workspace: 2,
};
const onCreateMock = jest.fn();
const onUpdateMock = jest.fn();
const clearErrors = jest.fn();
const validate = jest.fn();
const formState = "CREATE";
const workspace: any = [{ id: 2, name: "Default" }];

let createForm: RenderResult<typeof queries>;

beforeEach(() => {
	act(() => {
		createForm = render(
			<CreateProject
				clearErrors={clearErrors}
				initialValues={intialFormValue}
				validate={validate}
				formState={PROJECT_ACTIONS.CREATE}
				onCreate={onCreateMock}
				onUpdate={onUpdateMock}
				workspace={workspace}
			/>
		);
	});
});

describe("Create project Form", () => {
	test("should have a name field", async () => {
		let nameField = await createForm.getByTestId("createProjectName");
		expect(nameField).toBeInTheDocument();
	});
	test("should have a short-name field", async () => {
		let shortNameField = await createForm.getByTestId("createProjectShortName");
		expect(shortNameField).toBeInTheDocument();
	});
	test("should have a workspace field", async () => {
		let workspacefield = await createForm.getByTestId("createProjectWorkspace");
		expect(workspacefield).toBeInTheDocument();
	});
	test("should have a description field", async () => {
		let descriptionField = await createForm.getByTestId("createProjectDescription");
		expect(descriptionField).toBeInTheDocument();
	});
	test("should have a submit button", async () => {
		let submitButton = await createForm.getByTestId("createProjectSubmit");
		expect(submitButton).toBeInTheDocument();
	});
});

it("matches project form snapshot", () => {
	const tree = renderer
		.create(
			<CreateProject
				clearErrors={clearErrors}
				initialValues={intialFormValue}
				validate={validate}
				formState={PROJECT_ACTIONS.CREATE}
				onCreate={onCreateMock}
				onUpdate={onUpdateMock}
				workspace={workspace}
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
