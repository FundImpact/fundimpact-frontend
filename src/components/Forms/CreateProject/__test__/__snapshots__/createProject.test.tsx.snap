// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches project form snapshot 1`] = `
<div
  onChange={[MockFunction]}
>
  <form
    action="#"
    className="makeStyles-form-36"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="createProjectName"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
          >
            Project Name
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-38 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-40"
              >
                <span>
                  Project Name
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="createProjectShortName"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
          >
            Short Name
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="short_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-38 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-40"
              >
                <span>
                  Short Name
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-37"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink={true}
            id="demo-simple-select-outlined-label"
          >
            Choose Workspace
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            data-testid="createProjectWorkspace"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="demo-simple-select-outlined"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Default
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              name="workspace"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value={2}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-38 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-40 PrivateNotchedOutline-legendNotched-41"
              >
                <span>
                  Choose workspace
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="createProjectDescription"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Description.. ( Optional )
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
            onClick={[Function]}
          >
            <textarea
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
              disabled={false}
              name="description"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={3}
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-38 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-40"
              >
                <span>
                  Description.. ( Optional )
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        />
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            className="MuiBox-root MuiBox-root-43"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              data-testid="createProjectSubmit"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
